### PR TITLE
Adding KGGenerator

### PR DESCRIPTION
**Pull request by thinhngo-x**
- TextSplitter implemented but due to ollama does not have tokenizer for llama3, we cannot count the number of tokens of a prompt. Therefore, I set a hard-limit for number of words in a document instead, limited to 4K words (half of llama3 context window length - 8K).
- Entity resolution: a naive solution for now, merging all entities with the same ID.

### CODE REVIEW

File: backend/kg_generation/prompt_processor.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            - Entity IDs must be lowercase, no special characters or spaces, with words separated by "_".
+            - NO YAPPING before or after your answers. DO NOT add any comment or notes in your answers.
+            - Format your answers to strictly follow the rules in the example below.
+        Example:
+        Data: Temperature measurements taken from the Pacific Ocean show an increase due to global warming.
+        Types: [Process, Property]
+        Nodes: ["temperature_measurements", "Property", {}], ["pacific_ocean", "Realm", {"text": "Pacific Ocean"}], ["increase", "Trend", {"direction": "increase"}], ["global_warming", "Phenomenon", {"text": "Global Warming"}]
+        Relationships: ["temperature_measurements", "hasLocation", "pacific_ocean", {}], ["temperature_measurements", "hasTrend", "increase", {}], ["increase", "isCausedBy", "global_warming", {}]
+        """
+        self.user_prompt_template = PromptTemplate.from_template(
```
Comment (Viet1004) [inline backend/kg_generation/prompt_processor.py:34]: What does Types in user_prompt_template do? Specifying existing labels?

File: backend/kg_generation/generator.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        return self.prompt_processor.text_splitter
+
+    def from_text(self, text, labels=None):
+        prompt_messages = self.prompt_processor.create_prompt(text, labels)
+        llm_ans = self.llm.run(prompt_messages)
+        kg = self.prompt_processor.process_answer(llm_ans)
+        return kg
+
+    def add_kg(self, kg: KnowledgeGraph):
+        if "relationships" not in self.kg:
```
Comment (Viet1004) [inline backend/kg_generation/generator.py:38]: "relationships" always exists in self.kg by initialization?

File: backend/kg_generation/prompt_processor.py (hunk 1 above)
Comment (thinhngo-x) [reply backend/kg_generation/prompt_processor.py:34]: Yes. To reduce the number of entity types.

File: backend/kg_generation/generator.py (hunk 2 above)
Comment (thinhngo-x) [reply backend/kg_generation/generator.py]: removed.

**Review by Viet1004 — APPROVED** [review]

**Pull request merged**